### ISSUE TITLE
Document cutscene format, type 0x11

### ISSUE THREAD

**Issue by dertseha** [inkyblackness/ss-specs#28]
Document the remainder of the MOVI chunks for cutscenes (audio logs are already covered).

**Comment by dertseha** [comment]
The so far unknown 32bit field at `0014` in the MOVI header contains the same number for identical videos (for example, all the intro videos across all languages and formats have 0x000A0000).

I changed this field to various values, without any visible or audible effect. I guessed it might be related to which music plays or initial color settings for subtitles... nothing.

**Comment by dertseha** [comment]
The palette and subtitle entries are simple, and it showed me again how important it is to verify and re-verify with the actual files.

The original document of TTSHP reads (to me) as if the 0x4C palette entry re-uses the 0x04 entry. While it has the same timestamp and same data offset, its length is actually 0! So it is not re-using the palette data of the following 0x04 entry. I wrote that bit now clearer.
I suspect the 0x4C entry is a "reset palette" entry, though the actual meaning is most likely irrelevant. As long as both entries come in pairs, all is fine.

Next up are the actual video frame compressions.

**Comment by dertseha** [comment]
I had a go at the high-resolution compression. Using the documentation and the code from TSSHP as a guide, I now have a working prototype implementation from which I can derive a new documentation.

But the current code needs a few hacks. While the images turn out nearly OK, there are a few buffer overruns, which Go immediately detects. I'd guess TSSHP had the same issues but never detected because of unchecked memory access. This mainly concerns the last tile of a few frames (the bottom right 16 pixel), which, with the current prototype code, stay transparent.
Using the original video as a reference, I can't detect similar errors so I suspect some default/special handling in the corner cases.

**Comment by dertseha** [comment]
Cleaning up the hacks and finding a common thing brought the conclusion: The people designing the compression tried to remove any byte possible - even if it meant only saving a handful of bytes. In this case: if the decoder still wants to read one or two bytes past the end of the available data stream, simply return 0 values for those missing. Why should we store zeroes anyway? :)

The affected frames turn out all right now.

**Comment by dertseha** [comment]
Going through the full frames I found an issue with a few pixel after a scene change. In this case in the intro when Shodan gains conscience - see attached image, fram 1079.
![video_full_0bd6_1079](https://cloud.githubusercontent.com/assets/1508944/12117065/6ae63d4a-b3be-11e5-8759-1ad8689981bd.png)
The glaring pixel are at exact positions where the previous frame (hacker and Diego) had some specific corners. My guess is that the scene change does not completely wipe out all pixel and the remaining ones have different colour values due to palette change.

Looking at a screenshot from the game I can see similar pixel error, though they are not as pronounced (darker values).

In the frame by itself (1079 without previous ones accumulated) the pixel in question are marked as transparent.
TSSHP documentation notes that for type 0 control words, 0x00 pixel are not meant to be transparent. Though I doubt there was a special handling there.
I'll test with the prototype decoder.

**Comment by dertseha** [comment]
That one special handling of type 0 does not affect the outcome of this - Good, this means there is no special case for the pixel handling.

Which leaves us with the problem at hand. Looking at the pixel in the game screenshot, the pixel are at the same positions, but have far better color values, more in line of the new palette. I am now suspecting the video engine is even as much sophisticated to re-color all pixel at a palette change to chose palette indices with color values as close as possible to the new palette.
Another investigation.

**Comment by dertseha** [comment]
The engine most likely was not as sophisticated as I'd suspect here. The pixel errors that I saw in the screenshot were from a later stage and most likely leftovers from the whirling things floating around (which have about the same color as the single pixel have). This could rather be an error of the original encoder.

I catched a screenshot an an earlier stage of the scene where the pixel in question are (nearly) all (nearly) black.
It will most likely be far more simple: At a palette change, the entire screen is wiped with palette index 0x00, resulting in black. This color should not be transparent, as I am marking it in the test application and ending up in the file.

Still, an interesting find: scene changes do not always clear the entire frame buffer with a full frame. A decoder has to wipe it by itself as well.
